File: spryntax/website_static_build_tools/alg_creator/algs-c/depthfirstsearch.c
// Depth-First Search (DFS) using recursion
#include <stdio.h>
#include <stdlib.h>
// Function to perform DFS traversal
void dfs(int graph[][10], int visited[], int node, int n) {
	// Mark the node as visited
	visited[node] = 1;
	printf("%d ", node);

	// Recursively visit each adjacent node
	for (int i = 0; i < n; i++) {
		if (graph[node][i] == 1 && !visited[i]) {
			dfs(graph, visited, i, n);
		}
	}
}
int main() {
	// Adjacency matrix representation of the graph
	int graph[10][10] = {
		{0, 1, 1, 0, 0, 0},
		{0, 0, 0, 1, 1, 0},
		{0, 0, 0, 0, 0, 1},
		{0, 0, 0, 0, 0, 0},
		{0, 0, 0, 0, 0, 1},
		{0, 0, 0, 0, 0, 0}
	};
	// Array to track visited nodes
	int visited[10] = {0};
	// Start DFS traversal from node 0
	dfs(graph, visited, 0, 6);
	return 0;
}

File: spryntax/website_static_build_tools/alg_creator/algs-py/depthfirstsearch.py
# Depth-First Search (DFS) using recursion
# Function to perform DFS traversal
def dfs(graph, node, visited):
    # Mark the node as visited
    if node not in visited:
        print(node, end=" ")
        visited.add(node)
        # Recursively visit each adjacent node
        for neighbor in graph[node]:
            dfs(graph, neighbor, visited)
# Sample graph represented as an adjacency list
graph = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': ['F'],
    'F': []
}

# Set to track visited nodes
visited = set()

# Start DFS traversal from node 'A'
dfs(graph, 'A', visited)
